Lombiq.Analyzers.PowerShell/Rules/Measure-VariableNameFormat/Measure-VariableNameFormat.psm1

```diff
@@ -0,0 +1,344 @@
+﻿<#
+.SYNOPSIS
+    Detects inconsistencies in the casing of parameter names and variable names (including automatic variables).
+.DESCRIPTION
+    Raises the following warnings regarding the casing of parameter names and variable names (including automatic
+    variables):
+    - PSAvoidUsingUnnecessaryBracesInVariableNames: Variable names should not use unnecessary braces.
+    - PSUseCorrectAutomaticVariableNames: Automatic variables should be used exactly as they are documented:
+      https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables.
+    - PSUseCorrectParameterNameCasing: Parameter names should start with an uppercase letter.
+    - PSUseCorrectVariableNameCasing: Variable names should start with a lowercase letter.
+    - PSUseParametersAsDeclared: Parameters should be used with exactly as they were declared.
+
+    When fixing warnings, work through the rules in the order they are listed above, because violating
+    PSAvoidUsingUnnecessaryBracesInVariableNames and/or PSUseCorrectParameterNameCasing, while referencing that
+    parameter with the correct casing or format, will also raise a PSUseParametersAsDeclared warning. The latter will
+    disappear after fixing the former(s).
+.EXAMPLE
+    Measure-VariableNameFormat -Ast $Ast
+.INPUTS
+    [System.Management.Automation.Language.ScriptBlockAst]
+.OUTPUTS
+    [Microsoft.Windows.Powershell.ScriptAnalyzer.Generic.DiagnosticRecord[]]
+#>
+
+using namespace System.Management.Automation.Language
+
+Import-Module (Join-Path (Split-Path -Path $MyInvocation.MyCommand.Path) '..\AstFunctions.ps1') -Force
```

Comment:
Can you replace the `'..\AstFunctions.ps1'` with two separate parameters so you don't pass OS-specific path separator to `Join-Path`? Also I don't think you need the apostrophes.

Reply:
Should it be `'../AstFunctions.ps1'` instead? Because Windows doesn't care. BTW even this works on Ubuntu, because PS takes care of the conversion.
I put path strings intentionally (even if there are no spaces in them) in apostrophes/doublequotes for syntax highlighting.

Reply:
It should be either
```sh
Join-Path (Split-Path -Path $MyInvocation.MyCommand.Path) '..' 'AstFunctions.ps1'
```
or
```sh
Join-Path (Split-Path -Path $MyInvocation.MyCommand.Path) .. AstFunctions.ps1
```
since you are supposed to pass individual path segments to `Join-Path`. In practice it will also replace your path separator with the OS-appropriate one, but that's implicit error handling. Also it's not very aesthetic.

My preference is for the latter. Having the syntax highlighter color this as string when it's a path doesn't seem super useful to me.

Reply:
OIC! I prefer the first, because these are literal strings and I find highlighting them differently compared to other syntactical elements useful.

Reply:
Sure, go for it if you find it useful.